crates/bevy_ui/src/widget/slider.rs

```diff
@@ -0,0 +1,163 @@
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_ecs::query::With;
+use bevy_ecs::reflect::ReflectComponent;
+use bevy_ecs::schedule::SystemLabel;
+use bevy_ecs::system::{Query, Res};
+use bevy_hierarchy::Children;
+use bevy_input::prelude::MouseButton;
+use bevy_input::touch::Touches;
+use bevy_input::Input;
+use bevy_reflect::std_traits::ReflectDefault;
+use bevy_reflect::Reflect;
+use thiserror::Error;
+
+use crate::{Interaction, Node, RelativeCursorPosition, Style, Val};
+
+/// Describes the slider-specific value, such as max and min values and step
+#[derive(Component, Debug, Clone, Copy, Reflect)]
+#[reflect(Component, Default)]
+pub struct Slider {
+    min: f32,
+    max: f32,
+    step: f32,
+    value: f32,
+}
+
+impl Default for Slider {
+    fn default() -> Self {
+        Self {
+            min: 0.,
+            max: 100.,
+            step: 0.,
```

Comment:
Should default for Step be something other than 0? I would expect 1 to be more useful than 0?

Reply:
The `step` represents by how much we round up the slider value. I chose 0 for default since it means no rounding, which I think is the desired effect in most cases.